src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -169,142 +198,144 @@ void EnSth_Destroy(Actor* thisx, PlayState* play) {
+    Message_StartTextbox(play, nextTextId, &this->actor);
 }
 
-void func_80B67348(EnSth* this, PlayState* play) {
-    u16 phi_a1;
+void EnSth_PostOceanspiderhouseReward(EnSth* this, PlayState* play) {
+    u16 nextTextId;
 
     SkelAnime_Update(&this->skelAnime);
 
     if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
         this->actor.flags &= ~ACTOR_FLAG_10000;
-        this->actionFunc = func_80B67540;
+        this->actionFunc = EnSth_HandleOceansideSpiderHouseConversation;
 
-        switch (this->actor.home.rot.z) {
-            case 8:
-            case 9:
-                phi_a1 = 0x1137;
+        switch (STH_GI_ID(&this->actor)) {
+            case GI_WALLET_ADULT:
+            case GI_WALLET_GIANT:
+                nextTextId = 0x1137; // I'm just glad it was something you needed
                 break;
 
-            case 6:
-                phi_a1 = 0x1138;
+            case GI_RUPEE_SILVER:
+                // unused code, as he only gives wallet, purple and red, silver is never assigned to STH_GI_ID
+                nextTextId = 0x1138; // That's my life's fortune
                 break;
 
-            case 5:
-                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_13_40)) {
-                    phi_a1 = 0x113D;
+            case GI_RUPEE_PURPLE:
+                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_OCEANSIDE_WALLET_UPGRADE)) {
+                    nextTextId = 0x113D; // That's my life's savings
                 } else {
-                    phi_a1 = 0x113C;
+                    nextTextId = 0x113C; // That's my life's savings (2)
                 }
                 break;
 
             default:
-                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_13_40)) {
-                    phi_a1 = 0x1142;
+                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_OCEANSIDE_WALLET_UPGRADE)) {
+                    nextTextId = 0x1142; // Well, thats all I have on me
                 } else {
-                    phi_a1 = 0x1141;
+                    nextTextId = 0x1141; // Well, thats all I have on me
                 }
                 break;
         }
-        Message_StartTextbox(play, phi_a1, &this->actor);
+        Message_StartTextbox(play, nextTextId, &this->actor);
     } else {
         func_800B8500(&this->actor, play, 1000.0f, 1000.0f, PLAYER_IA_MINUS1);
     }
 }
 
-void func_80B67458(EnSth* this, PlayState* play) {
+void EnSth_GiveOceanspiderhouseReward(EnSth* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
 
     if (Actor_HasParent(&this->actor, play)) {
         this->actor.parent = NULL;
-        this->actionFunc = func_80B67348;
+        this->actionFunc = EnSth_PostOceanspiderhouseReward;
         this->actor.flags |= ACTOR_FLAG_10000;
         func_800B8500(&this->actor, play, 1000.0f, 1000.0f, PLAYER_IA_MINUS1);
         if (CURRENT_DAY == 3) {
-            func_80B670A4(this, 6);
+            EnSth_SwitchAnimation(this, STH_ANIM_PLEAD);
```

Comment:
```suggestion
            EnSth_ChangeAnim(this, STH_ANIM_PLEAD);
```
`ChangeAnim` is the common function name used for this